src/Cli/dotnet/commands/dotnet-workload/WorkloadUtilities.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Deployment.DotNet.Releases;
+
+namespace Microsoft.DotNet.Workloads.Workload
+{
+    internal class WorkloadUtilities
+    {
+        internal static int VersionCompare(string first, string second)
+        {
+            if (first.Equals(second))
+            {
+                return 0;
+            }
+
+            var firstDash = first.IndexOf('-');
+            var secondDash = second.IndexOf('-');
+            firstDash = firstDash < 0 ? first.Length : firstDash;
+            secondDash = secondDash < 0 ? second.Length : secondDash;
+
+            var firstVersion = new Version(first.Substring(0, firstDash));
+            var secondVersion = new Version(second.Substring(0, secondDash));
+
+            var comparison = firstVersion.CompareTo(secondVersion);
+            if (comparison != 0)
+            {
+                return comparison;
+            }
+
+            var modifiedFirst = "1.1.1" + (firstDash == first.Length ? string.Empty : first.Substring(firstDash));
```

Comment:
You could just call the ctor overload for ReleaseVersion: `ReleaseVersion(int major, int minor, int patch, string prerelease = null, string buildMetadata = null)`, you'll have a proper value to compare and avoid creating two strings and then creating versions to compare.

Reply:
I cannot use that overload because it doesn't handle 4-part version numbers. The error this is trying to address is due to trying to create a ReleaseVersion from the original string where it happens to be 9.0.100.1.

Reply:
I'm suggesting that you do the following. Right now you're creating a string and passing that to ReleaseVersion constructor, so you have something that will parse correctly.

```
modifiedFirst = new ReleaseVersion(1, 1, 1, (firstDash == first.Length ? string.Empty : first.Substring(firstDash));

return modifiedFirst.CompareTo(modifiedSecond)
```

Reply:
Ah, I misunderstood. Yeah; that should work and will save a couple allocations 🙂 Happy to make that change.